external/fv3fit/fv3fit/emulation/models/transformed_model.py

```diff
@@ -1,9 +1,116 @@
 from typing import Mapping
 from fv3fit.keras.adapters import ensure_dict_output
 from fv3fit.emulation.transforms import TensorTransform
+from fv3fit.emulation.types import TensorDict
 import tensorflow as tf
 
-__all__ = ["transform_model"]
+__all__ = ["transform_model", "TransformedModel"]
+
+
+class TransformedModel(tf.keras.Model):
```

Comment:
I also considered simply saving separate keras Functional models for the forwards and backwards transformations, but ended up using a custom model instead. The use of tf.functions below lead to some complexity down the road, since they can "memorize" the wrong inputs.